cmd/mattermost/commands/webhook_test.go

```diff
@@ -49,3 +49,38 @@ func TestListWebhooks(t *testing.T) {
 	}
 
 }
+
+func TestCreateIncomingWebhook(t *testing.T) {
+	th := api4.Setup().InitBasic().InitSystemAdmin()
+	defer th.TearDown()
+
+	th.App.UpdateConfig(func(cfg *model.Config) { cfg.ServiceSettings.EnableIncomingWebhooks = true })
+	th.App.UpdateConfig(func(cfg *model.Config) { cfg.ServiceSettings.EnableOutgoingWebhooks = true })
+	th.App.UpdateConfig(func(cfg *model.Config) { cfg.ServiceSettings.EnablePostUsernameOverride = true })
+	th.App.UpdateConfig(func(cfg *model.Config) { cfg.ServiceSettings.EnablePostIconOverride = true })
+
+	defaultRolePermissions := th.SaveDefaultRolePermissions()
+	defer func() {
+		th.RestoreDefaultRolePermissions(defaultRolePermissions)
+	}()
+	th.AddPermissionToRole(model.PERMISSION_MANAGE_WEBHOOKS.Id, model.TEAM_ADMIN_ROLE_ID)
+	th.RemovePermissionFromRole(model.PERMISSION_MANAGE_WEBHOOKS.Id, model.TEAM_USER_ROLE_ID)
+
+	description := "myhookinc"
+	CheckCommand(t, "webhook", "create-incoming", "--channel", th.BasicChannel.Id, "--user", th.BasicUser.Id, "--description", description)
+
+	webhooks, err := th.App.GetIncomingWebhooksPage(0, 1000)
+	if err != nil {
+		t.Fatal("unable to retrieve incoming webhooks")
+	}
+
+	found := false
+	for _, webhook := range webhooks {
+		if webhook.Description == description {
+			found = true
+		}
+	}
+	if !found {
+		t.Fatal("Failed to create incoming webhook")
+	}
+}
```

Comment:
It would be great to have tests for the common expected failure modes: missing required fields, or invalid channel/user specified. One way to do this while minimising the amount of boiler-plate code written is to use a line like this to test each case where the command should fail:

```
require.Error(t, RunCommand(t, "channel", "remove", th.BasicTeam.Name+":doesnotexist", th.BasicUser2.Email))
```

Reply:
Added cases for expected failure paths